fhd_core/calibration/vis_cal_polyfit.pro

```diff
@@ -1,34 +1,39 @@
-    digital_gain_jump_polyfit=digital_gain_jump_polyfit,mode_debug=mode_debug,_Extra=extra
-
-IF Keyword_Set(cal_cable_reflection_fit) OR Keyword_Set(cal_cable_reflection_correct) THEN cal.mode_fit=1.
-cal_mode_fit=cal.mode_fit
-IF Tag_exist(cal,"amp_degree") THEN amp_degree = cal.amp_degree ELSE amp_degree=2
-IF Tag_exist(cal,"phase_degree") THEN phase_degree = cal.phase_degree ELSE phase_degree=1
-
-n_pol=cal.n_pol
-n_freq=cal.n_freq
-n_tile=cal.n_tile
-freq_arr=cal.freq
-IF N_Elements(obs) GT 0 THEN freq_use=where((*obs.baseline_info).freq_use,nf_use) ELSE freq_use=lindgen(n_freq)
-IF N_Elements(obs) GT 0 THEN tile_use=where((*obs.baseline_info).tile_use,nt_use) ELSE tile_use=lindgen(n_tile)
-IF Keyword_Set(cal_neighbor_freq_flag) THEN BEGIN
+    cal_reflection_hyperresolve=cal_reflection_hyperresolve,cal_reflection_mode_theory=cal_reflection_mode_theory,$
+    cal_reflection_mode_file=cal_reflection_mode_file,cal_reflection_mode_delay=cal_reflection_mode_delay, $
+    no_phase_calibration=no_phase_calibration,digital_gain_jump_polyfit=digital_gain_jump_polyfit,_Extra=extra
+    
+  IF Keyword_Set(cal_reflection_mode_theory) OR Keyword_Set(cal_reflection_mode_file) $
+    OR Keyword_Set(cal_reflection_mode_delay) OR Keyword_Set(cal_reflection_hyperresolve) AND $
+    (cal.mode_fit EQ 0) THEN cal.mode_fit=1.
+  cal_mode_fit=cal.mode_fit
+  IF (cal_mode_fit EQ 1) AND keyword_set(cal_reflection_mode_theory) then $
+    if (abs(cal_reflection_mode_theory) GT 1) then cal_mode_fit = cal_reflection_mode_theory
+  IF Tag_exist(cal,"amp_degree") THEN amp_degree = cal.amp_degree ELSE amp_degree=2
+  IF Tag_exist(cal,"phase_degree") THEN phase_degree = cal.phase_degree ELSE phase_degree=1
+  
+  n_pol=cal.n_pol
+  n_freq=cal.n_freq
+  n_tile=cal.n_tile
+  freq_arr=cal.freq
+  IF N_Elements(obs) GT 0 THEN freq_use=where((*obs.baseline_info).freq_use,nf_use) ELSE freq_use=lindgen(n_freq)
+  IF N_Elements(obs) GT 0 THEN tile_use=where((*obs.baseline_info).tile_use,nt_use) ELSE tile_use=lindgen(n_tile)
+  IF Keyword_Set(cal_neighbor_freq_flag) THEN BEGIN
     freq_use=(*obs.baseline_info).freq_use
     freq_flag=where(freq_use EQ 0,nf_flag)
     IF nf_flag GT 0 THEN BEGIN
-        FOR fi=0L,nf_flag-1 DO freq_use[((freq_flag[fi]-cal_neighbor_freq_flag)>0):((freq_flag[fi]+cal_neighbor_freq_flag)<(n_freq-1))]=0
+      FOR fi=0L,nf_flag-1 DO freq_use[((freq_flag[fi]-cal_neighbor_freq_flag)>0):((freq_flag[fi]+cal_neighbor_freq_flag)<(n_freq-1))]=0
     ENDIF
     freq_use=where(freq_use,nf_use)
-ENDIF
-
-c_light=299792458.
-i_comp=Complex(0,1)
-cal_return=cal
-FOR pol_i=0,n_pol-1 DO cal_return.gain[pol_i]=Ptr_new(*cal.gain[pol_i]) ;essential, to keep original cal gains from being overwritten!
-
-IF Keyword_Set(cal_step_fit) THEN BEGIN
+  ENDIF
+  
+  c_light=299792458.
+  i_comp=Complex(0,1)
+  cal_return=cal
+  FOR pol_i=0,n_pol-1 DO cal_return.gain[pol_i]=Ptr_new(*cal.gain[pol_i]) ;essential, to keep original cal gains from being overwritten!
```

Comment:
I think this could be simplified to `cal_return.gain=Pointer_copy(cal.gain)`

Reply:
done